### PR TITLE
593 studio refactor

### PR DESCRIPTION
**Pull request by dhaneshnm**
-  Refactor studio to avoid unnecessary re-render. 
- Move Gallery view and related routing to separate components.
- Add an abort in usePlugin to fix memory leak issue. This is happening when you open and close a resource from studio, DashBoardResultsContainer.

Fixes  BlueBrain/nexus/issues/1593

### CODE REVIEW

File: src/shared/hooks/usePlugins.ts [hunk 1]
```diff
@@ -52,13 +56,14 @@ export default function usePlugins() {
...
         )
-        .catch(error =>
+        .catch(error => {
           setManifest({
             error,
             loading: false,
             data: null,
-          })
-        );
+          });
```
Comment (dhaneshnm) [inline src/shared/hooks/usePlugins.ts:64]: May be we should make aborting a standard practice across all components that does API calls in UseEffect.

**Comment by dhaneshnm** [comment]
Screenshot 2020-10-08 at 09 17 58

File: src/shared/views/GalleryView.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  // a query param from any app path
+  const self = getUrlParameter('_self');
+  if (self && self !== '') {
+    nexus
+      .httpGet({
+        path: self,
+        headers: { Accept: 'application/json' },
+      })
+      .then((resource: Resource) => {
+        console.log('here ..');
```
Comment (Stafeeva) [inline src/shared/views/GalleryView.tsx:33]:
```suggestion
```

**Review by Stafeeva — APPROVED** [review]
⭐

File: src/shared/hooks/usePlugins.ts [hunk 3]
```diff
@@ -52,13 +56,14 @@ export default function usePlugins() {
...
+          return () => abort.abort();
```
Comment (kenjinp) [inline src/shared/hooks/usePlugins.ts:65]: AbortController requires you pass in a signal for it to work: [https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort)

**Review by kenjinp — CHANGES_REQUESTED** [review]

**Review by Stafeeva — APPROVED** [review]

**Review by kenjinp — APPROVED** [review]

**Pull request merged**